src/ngMock/angular-mocks.js

```diff
@@ -556,9 +556,16 @@ angular.mock.$IntervalProvider = function() {
      */
     $interval.flush = function(millis) {
       now += millis;
-      while (repeatFns.length && repeatFns[0].nextTime <= now) {
+      while (repeatFns.length && (repeatFns[0].nextTime <= now || !repeatFns[0].nextTime)) {
         var task = repeatFns[0];
         task.fn();
+        // when a delay of zero is given
+        // it has to be replaced with one
+        // to avoid running the while loop
+        // one more time than the specified
+        if (!task.delay) {
+          task.delay = task.nextTime = 1;
```

Comment:
Actually, I don't think this will work if this `0` delay timer was not added in the beginning (i.e. if there was a flush before adding it). E.g.:

```js
var counter = 0;
$interval.flush(1000);
$interval(function() { counter++; });
$interval.flush(1000);

expect(counter).toBe(1000);   // <-- Will probably be 2000
```

Also, due to how `repeatFns.nextTime` is set up initially (`nextTime: now + delay`), using leaving the delay unspecified and using `0` would behave differently.

I think we should change both `$interval` and `$interval.flush` to something like:

```js
$interval = function(fn, delay, ...) {
  ...
  repeatFns.push({
    nextTime: (now + (delay || 0)),
    delay: delay || 1,
    ...
  });
```

```js
$interval.flush = function(millis) {
  var before = now;
  now += millis;
  while (repeatFns.length && repeatFns[0].nextTime <= now) {
    ...
    if (task.nextTime === before) {
      // This can only happen the first time a zero-delay interval get triggered.
      task.nextTime++;
    }
    task.nextTime += task.delay;
    ...
```

Then we need to have more tests that verify the correct behavior of zero-delay intervals (with both `delay: 0` and `delay: undefined`) for at least the following cases: 

- Interval being added when `now` > 0.
- Calling `flush(0)` (should only trigger zero-delay intervals and only trigger them once).
- Calling `flush(0)` for a second time (should not trigger any intervals).